### PR TITLE
t/mojo/dynamic_methods.t now uses subtests.

### PR DESCRIPTION
**Pull request by kberov**
### Summary
t/mojo/dynamic_methods.t now uses subtests.

### Motivation
Contribute to the Subtests iniciative
### References
#1520

### CODE REVIEW

File: t/mojo/dynamic_methods.t [hunk 1]
```diff
@@ -22,22 +22,24 @@ sub BUILD_DYNAMIC {
...
-ok !Mojo::TestDynamic->can('foo'), 'dynamic method is hidden';
-ok eval { $t1->foo; 1 }, 'foo called ok';
-cmp_ok $called_foo, '==', 1, 'called dynamic method';
-ok !eval { $t2->foo; 1 }, 'error calling foo on wrong object';
-
-# Garbage collection
-undef($t1);
-undef($t2);
-ok(!keys(%$dyn_methods), 'dynamic methods expired');
+my ($t1, $t2, $dyn_methods);
```
Comment (kraih) [inline t/mojo/dynamic_methods.t:25]: No, just no.

Comment (kberov) [reply t/mojo/dynamic_methods.t:25]: What would be expected to be done in this case?

**Review by kraih — CHANGES_REQUESTED** [review]
The point of subtests is to make the code cleaner and more readable.

**Comment by kberov** [comment]
Then this file does not need subtests. It is already readable enough.

**Pull request closed without merging**